File: Non-procedural (functional) programming/Solved/antidvorak/kombinace.pl
?- consult('functional.pl').

%kombinace se mi hodili furt

combinations(0,_,[[]]) :- !.
combinations(_, [], []) :- !.
combinations(N, [X | Xs], Ks) :-
    N > 0,
    combinations(N, Xs, K1s),
    M is N - 1,
    combinations(M, Xs, K2s),
    map(pushFront(X), K2s, K3s),
    append(K1s, K3s, Ks).

combination(N, Xs, Ys) :-
    combinations(N, Xs, Z),
    member(Ys, Z).

File: Non-procedural Programming/Solved/antihric/kombinace.hs
module Kombinace where

--kombinace se mi hodili furt

kombinace :: (Eq a, Integral b) => b -> [a] -> [[a]]
kombinace 0 _ = [[]]
kombinace _ [] = []
kombinace n (x : xs) | n > 0 = kombinace n xs ++ map (x:) (kombinace (n-1) xs)